Contributions: "give to the Guardian" header with coin icon

rtyley opened the pull request:
## What does this change?

Testing a "give to the Guardian" header, which replaces 'become a member' in the nav bar. Links to https://contribute.theguardian.com/

## Does this affect other platforms - Amp, Apps, etc?

Nope.

## Screenshots

With the coin
![image](https://cloud.githubusercontent.com/assets/52038/17407944/1fd4d5c4-5a61-11e6-9efd-af18eeb63981.png)

without (ie normal view):
![image](https://cloud.githubusercontent.com/assets/52038/17407927/0e41cbd2-5a61-11e6-866b-9edce75b98b4.png)


## Request for comment

cc @markjamesbutler @desbo 

<!--
*Does this PR meet the [contributing guidelines](https://github.com/guardian/frontend/blob/issue_pr_templates/.github/CONTRIBUTING.md#submission)?*
-->

markjamesbutler commented on common/app/conf/switches/ABTestSwitches.scala:
```diff
@@ -63,4 +63,14 @@ trait ABTestSwitches {
+    "ab-giraffe-header-20160802",
+    "Test effectiveness of header for driving contributions.",
+    owners = Seq(Owner.withGithub("markjamesbutler")),
+    safeState = On,
```
Let's change this to Off just in case.

gtrufitt commented on static/src/inline-svgs/contributions/coin.svg:
```diff
@@ -0,0 +1,39 @@
+<?xml version="1.0" encoding="utf-8"?>
+<!-- Generator: Adobe Illustrator 16.0.1, SVG Export Plug-In . SVG Version: 6.00 Build 0)  -->
```
You should run this through https://jakearchibald.github.io/svgomg/

rtyley replied:
Will do - I'm a bit surprised though, https://github.com/guardian/frontend/blob/master/docs/inline-svgs.md#image-preparation says that inline SVGs are run through SVGO anyway, but they don't seem to be?

gtrufitt replied:
Hmh. One to add to our list...

zeftilldeath commented:
Maybe the coin could be the full circle size with no outline around it. Would be larger, then and I think it looks a bit strange being a circle within a circle. Our roundel does the same.

NataliaLKB commented on static/src/javascripts/projects/common/modules/experiments/ab.js:
```diff
@@ -27,7 +28,8 @@ define([
     Giraffe,
```
is `Giraffe` and `giraffeHeader` the clearest names for this? I know that your team name is that, but your team name may change and this code may still be here

rtyley replied:
Yeah, I'd be happy to change 'giraffe'-> 'contributions', though this PR is only to add the new header, so shall I just fix the `giraffeHeader` for the time being?

NataliaLKB replied:
Yes, but please do rename to contributions in a separate PR 😄

gtrufitt commented on static/src/javascripts/projects/common/modules/experiments/tests/giraffe-header.js:
```diff
@@ -0,0 +1,72 @@
+        this.idealOutcome = '';
+        this.canRun = function () {
+            var pageObj = window.guardian.config.page;
+            return !(pageObj.isSensitive || pageObj.isLiveBlog || pageObj.isAdvertisementFeature) && pageObj.edition === 'UK';
```
Do you need to check `isSensitive` if you've set `showForSensitive` to` `false`?

rtyley replied:
https://github.com/guardian/frontend/blob/3194e47/static/src/javascripts/projects/common/modules/experiments/ab.js#L111-L113 seems to indicate that `showForSensitive` is keyed off `shouldHideAdverts`, which sounds reasonable - I'll take out the `pageObj.isSensitive` clause.

NataliaLKB commented on static/src/inline-svgs/contributions/coin.svg:
```diff
@@ -0,0 +1 @@
+<svg xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" width="18" height="18" viewBox="5 5 18 18"><defs><circle id="a" cx="14" cy="14" r="9"/></defs><use xlink:href="#a" overflow="visible" fill="#FB0"/><clipPath id="b"><use xlink:href="#a" overflow="visible"/></clipPath><circle clip-path="url(#b)" fill="none" stroke="#FF9B0B" stroke-miterlimit="10" stroke-dasharray="2, ...
```
Please follow https://github.com/guardian/frontend/blob/master/docs/inline-svgs.md and get rid off all the `xmlns` things

gtrufitt commented on static/src/javascripts/projects/common/modules/experiments/tests/giraffe-header.js:
```diff
@@ -0,0 +1,72 @@
+                test: function () {
+                    writer('give to the Guardian', 'https://contribute.theguardian.com/uk?INTCMP=co_uk_head_give_coin');
+                },
+                success: function (complete) {
```
`success: completer` would be the same?

desbo replied:
since there's only one use I'd just define the function at the property

jfsoul replied:
but could still do the same to line 55, could be just `...'click', complete)`

gtrufitt commented on static/src/javascripts/projects/common/views/giraffe-header.html:
```diff
@@ -0,0 +1,12 @@
+                                    hide-until-mobile-landscape">
+    <a href="<%= linkHref %>"
+       id="giraffe__header-contribute-link"
+       data-link-name="give to the Guardian"
```
Might be worth making this a bit more descriptive eg: `header : give to the Guardian` to differentiate it from other place

piuccio commented:
looks kinda like the lidl logo, yellow circle on blue 😁

johnduffell commented on docs/ab-testing.md:
```diff
@@ -165,7 +165,7 @@ You can stop and start the test using our [switchboard](https://frontend.gutools
 To see the test in action locally: `localhost:9000/[articleId]#ab-[testName]=[variantName]`
 
-To see the tests you are part of using Google Chrome: open the Dev Console -> Resources -> Local Storage -> choose `http://[domain]` (e.g. localhost or www.theguardian.com)  -> check the `gu.ab.participations` row.
+To see the tests you are part of using Google Chrome: open the Dev Console -> Resources -> Local Storage -> choose `http://[domain]` (e.g. localhost or www.theguardian.com)  -> check the `gu.ab.participations` row. Alternatively, if you're using Chrome, use the [extension](https://chrome.google.com/webstore/detail/guardian-ab-tests/nehbenedinjacnhlkjbdneedibcagmno).
```
good one!

johnduffell commented on static/src/inline-svgs/contributions/coin.svg:
```diff
@@ -0,0 +1 @@
+<svg width="18" height="18" viewBox="5 5 18 18"><defs><circle id="a" cx="14" cy="14" r="9"/></defs><use overflow="visible" fill="#FB0"/><clipPath id="b"><use overflow="visible"/></clipPath><circle clip-path="url(#b)" fill="none" stroke="#FF9B0B" stroke-miterlimit="10" stroke-dasharray="2,2" cx="14" cy="14" r="7.615"/><g clip-path="url(#b)"><defs><path id="c" d="M11.64 14.604h.655v.593c0 .86-.296  ...
```
💰

johnduffell commented:
looks great, will be an interesting test, although I'm with zef on the logo, it looks a bit out of place for some reason.  would be interesting to do further testing on the exact wording too
👍 for the idea

NataliaLKB commented on common/app/conf/switches/ABTestSwitches.scala:
```diff
@@ -63,4 +63,14 @@ trait ABTestSwitches {
     exposeClientSide = true
   )
+
+  val ABGiraffeHeader20160802 = Switch(
```
Could we rename the test too?

rtyley commented:
I think I've addressed all the feedback in this PR (apart from the logo design itself - Alex B did sign-off on this, but obviously it's going to be worth iterating on) - can I proceed with merging?

NataliaLKB commented:
I have one more comment, and also agree with @zeftilldeath 's recommendation regarding the design. After that 👍
